akka-stream-testkit/src/test/scala/akka/stream/testkit/StreamTestKit.scala

```diff
@@ -1,177 +1,9 @@
-    val probe = TestProbe()
-
-    def expectSubscription(): Subscription = probe.expectMsgType[OnSubscribe].subscription
-    def expectEvent(event: SubscriberEvent): Unit = probe.expectMsg(event)
-    def expectNext(element: I): Unit = probe.expectMsg(OnNext(element))
-    def expectNext(e1: I, e2: I, es: I*): Unit = {
-      val all = e1 +: e2 +: es
-      all.foreach(e ⇒ probe.expectMsg(OnNext(e)))
-    }
-
-    def expectNext(): I = probe.expectMsgType[OnNext[I]].element
-    def expectComplete(): Unit = probe.expectMsg(OnComplete)
-
-    def expectError(cause: Throwable): Unit = probe.expectMsg(OnError(cause))
-    def expectError(): Throwable = probe.expectMsgType[OnError].cause
-
-    def expectSubscriptionAndError(cause: Throwable): Unit = {
-      val sub = expectSubscription()
-      sub.request(1)
-      expectError(cause)
-    }
-    def expectSubscriptionAndError(): Throwable = {
-      val sub = expectSubscription()
-      sub.request(1)
-      expectError()
-    }
-
-    def expectSubscriptionAndComplete(): Unit = {
-      val sub = expectSubscription()
-      sub.request(1)
-      expectComplete()
-    }
-
-    def expectNextOrError(element: I, cause: Throwable): Either[Throwable, I] = {
-      probe.fishForMessage(hint = s"OnNext($element) or ${cause.getClass.getName}") {
-        case OnNext(n)        ⇒ true
-        case OnError(`cause`) ⇒ true
-      } match {
-        case OnNext(n: I @unchecked) ⇒ Right(n)
-        case OnError(err)            ⇒ Left(err)
-      }
-    }
-
-    def expectNoMsg(): Unit = probe.expectNoMsg()
-    def expectNoMsg(max: FiniteDuration): Unit = probe.expectNoMsg(max)
-
-    def onSubscribe(subscription: Subscription): Unit = probe.ref ! OnSubscribe(subscription)
-    def onNext(element: I): Unit = probe.ref ! OnNext(element)
-    def onComplete(): Unit = probe.ref ! OnComplete
-    def onError(cause: Throwable): Unit = probe.ref ! OnError(cause)
-  }
-
-  object PublisherProbe {
-    def apply[I]()(implicit system: ActorSystem): PublisherProbe[I] = new PublisherProbe[I]()
-  }
-
-  class PublisherProbe[I]()(implicit system: ActorSystem) extends Publisher[I] {
-    val probe: TestProbe = TestProbe()
-
-    def subscribe(subscriber: Subscriber[_ >: I]): Unit = {
-      val subscription: PublisherProbeSubscription[I] = new PublisherProbeSubscription[I](subscriber, probe)
-      probe.ref ! Subscribe(subscription)
-      subscriber.onSubscribe(subscription)
-    }
-
-    def expectSubscription(): PublisherProbeSubscription[I] =
-      probe.expectMsgType[Subscribe].subscription.asInstanceOf[PublisherProbeSubscription[I]]
-
-    def expectRequest(subscription: Subscription, n: Int): Unit = probe.expectMsg(RequestMore(subscription, n))
-
-    def expectNoMsg(): Unit = probe.expectNoMsg()
-    def expectNoMsg(max: FiniteDuration): Unit = probe.expectNoMsg(max)
-
-    def getPublisher: Publisher[I] = this
-  }
+object Utils {
```

Comment:
It ocnly contains the `TE`, so moving it to `StreamTestKit` or something like that would work well I think :)